hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/DropHoodieTableCommand.scala

```diff
@@ -85,25 +91,42 @@ case class DropHoodieTableCommand(
   }
 
   private def dropHiveDataSourceTable(
-        sparkSession: SparkSession,
-        table: CatalogTable,
-        ifExists: Boolean,
-        purge: Boolean): Unit = {
+       sparkSession: SparkSession,
+       hoodieCatalogTable: HoodieCatalogTable): Unit = {
+    val table = hoodieCatalogTable.table
     val dbName = table.identifier.database.get
     val tableName = table.identifier.table
+
     // check database exists
     val dbExists = sparkSession.sessionState.catalog.databaseExists(dbName)
     if (!dbExists) {
       throw new NoSuchDatabaseException(dbName)
     }
-    // check table exists
-    if (!sparkSession.sessionState.catalog.tableExists(table.identifier)) {
-      throw new NoSuchTableException(dbName, table.identifier.table)
+
+    if (HoodieTableType.MERGE_ON_READ == hoodieCatalogTable.tableType && purge) {
+      val snapshotTableName = hoodieCatalogTable.tableName + SUFFIX_SNAPSHOT_TABLE
+      val roTableName = hoodieCatalogTable.tableName + SUFFIX_READ_OPTIMIZED_TABLE
+
+      dropHiveTable(sparkSession, dbName, snapshotTableName)
+      dropHiveTable(sparkSession, dbName, roTableName)
+      dropHiveTable(sparkSession, dbName, hoodieCatalogTable.tableName, purge)
```

Comment:
why still need the original tableName?

Reply:
> why still need the original tableName?

Because purge deletes the directory of the original table.

Reply:
let's unify the `tableName` here.